x-pack/plugins/lens/public/editor_frame_service/editor_frame/expression_helpers.ts

```diff
@@ -79,6 +171,8 @@ export function buildExpression({
   searchSessionId?: string;
   dateRange: DateRange;
   nowInstant: Date;
+  // should be set to false when: 1. searchcontext changes (filters, query, timerange) or 2. refresh is clicked
+  canUseCache: boolean | undefined;
```

Comment:
for esaggs this seems irrelevant as samplingseed is set to different value when we refresh/update search context, so we can always pass `True`